sql-core/src/main/java/io/cdap/sql/core/SQLQuery.java

```diff
@@ -0,0 +1,113 @@
+package io.cdap.sql.core;
+
+import io.cdap.sql.core.enums.SQLReadableType;
+import io.cdap.sql.core.interfaces.Aliasable;
+import io.cdap.sql.core.interfaces.Filterable;
+import io.cdap.sql.core.interfaces.SQLReadable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SQLQuery implements Aliasable, Filterable, SQLReadable {
+  private SQLQuery(List<SQLColumn> selectCols, SQLFilter filter, SQLReadable from, String alias) {
+    selectColumns = selectCols;
```

Comment:
`selectColumns = Collections.unmodifiableList(selectCols)`

this way you're guaranteed that nobody who calls getColumns() can wrongly modify the list.

Reply:
Fixed